core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -608,9 +608,14 @@ private function addBackupNode(): NodeDefinition
                     ->scalarPrototype()->end()
                 ->end()
                 ->integerNode('keep_max')
-                    ->info('The maximum number of backups to keep.')
+                    ->info('The maximum number of backups to keep. Use 0 to keep all the backups forever.')
                     ->defaultValue(5)
                 ->end()
+                ->arrayNode('keep_periods')
+                    ->info('The latest backup plus the oldest of every configured period will be kept.')
+                    ->defaultValue([1, 7, 14, 30])
```

Comment:
Can we maybe support strings like `7d`, `2w`, `1m` here? I think this would make for a more meaningful config.

Reply:
That was real PITA because `DateInterval` instances are not comparable by default but I've found a smart workaround :P Implemented in https://github.com/contao/contao/pull/3850/commits/d0c63da6771500eb134222ea95a259459c81f52f

Reply:
Nice!